src/FileInputFieldWithList.js

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react'
+import propTypes from '@dhis2/prop-types'
+
+import { FileInputField } from './FileInputField.js'
+import { FileListItemWithRemove } from './FileInputFieldWithList/FileListItemWithRemove.js'
+import { statusPropType, sizePropType } from './common-prop-types.js'
+
+/* 
+NOTE:
+A JavaScript FileList instance is read-only, so we cannot add files to it
+FileList also doesn't have a .map method so by destructuring the FileList
+instance into an array we can add, remove and map
```

Comment:
So eventually we can remove the comment above, right?

Reply:
@ismay, I'm not sure I understand you question correctly. If you mean that this comment can be removed and replaced by JSdoc comments, then YES I agree.

Reply:
I might have misunderstood, but I thought that instead of using an array the intention was to move to using a FileList object (appending to it by creating a new FileList)? Or does that not apply to this section? That's why I thought that eventually this comment could be removed as well.

Reply:
@ismay thanks for clarifying. You are correct in thinking that we did consider switching to a `FileList` instance. However, after looking into this in more detail, I didn't seem it would offer any benefits when handling the form data, and at the same time it would complicate the component code a bit. So I kept things the way they were.

Reply:
Below is a copy of my full motivation which I provided to @varl on Slack.
> I’ve looked into that Array-of-File-instances VS FileList-instance stuff quickly and think we should keep using the current array-based approach. I’ve looked at how you can append several files to a `FormData` instance, and all I see is examples that loop over the FileList to append stuff using the bracket notation, i.e.
>
>```javascript
>formData.append('userpic[]', myFileInput.files[0], 'chris1.jpg');
>formData.append('userpic[]', myFileInput.files[1], 'chris2.jpg');
>```

> source: https://developer.mozilla.org/en-US/docs/Web/API/FormData/append

> ```javascript
> var ins = document.getElementById('fileToUpload').files.length;
> for (var x = 0; x < ins; x++) {
>    fd.append("fileToUpload[]", document.getElementById('fileToUpload').files[x]);
> }```

> https://stackoverflow.com/a/14908250/1143502

> Since it seems we need to append every file individually to the FormData, using an array is equally convenient as using a FileList instance. And since it is more convenient from the perspective of the `FileInputWithList` component to keep using an array, I would say we stick to that.

Reply:
Ah I see. Ok yeah, makes sense, thanks for explaining!